src/app/coursegrab/utils/authorize.py

```diff
@@ -26,3 +26,19 @@ def inner(*args, **kwargs):
         return f(user=user, session=session, *args, **kwargs)
 
     return inner
+
+def authorize_user_selective(f):
+    """
+    If no auth header is present, don't throw any error.
+    If auth header is present, verify the session.
+    """
+    @wraps(f)
+    def inner(*args, **kwargs):
+        auth_header = request.headers.get("Authorization")
+        if auth_header is None:
+            return f(*args, **kwargs)
+        bearer_token = auth_header.replace("Bearer ", "").strip()
+        user, session = verify_session(bearer_token)
+        return f(user=user, session=session, *args, **kwargs)
```

Comment:
why do you need to have a wrap thingie?

Reply:
wraps() basically allows the wrapper function (in this case, `inner`) to have the same property as the wrapped function (in this case, `f`) . Not super necessary, but helps with debugging and it's a common practice!